pkg/clients/kube/kube.go

```diff
@@ -0,0 +1,149 @@
+// Provides a client for interacting with jobs and pods through the KubeAPI in a single namespace
+// Generated by Claude 3.5 Sonnet
+package kube
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Client represents a Kubernetes Client
+type Client struct {
+	clientset kubernetes.Interface
+	namespace string
```

Comment:
I would rather callers pass the namespace into functions than have to create a new client for each namespace.